src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -52,6 +52,92 @@ Copyright (c) .NET Foundation. All rights reserved.
+        <RuntimeFrameworkVersion Condition=" $(RuntimeFrameworkVersion.Split('.').Length) == 1">$(RuntimeFrameworkVersion).0.0</RuntimeFrameworkVersion>
+        <RuntimeFrameworkVersion Condition=" $(RuntimeFrameworkVersion.Split('.').Length) == 2">$(RuntimeFrameworkVersion).0</RuntimeFrameworkVersion>
+
+        <NetCoreAppImplicitPackageVersion Condition="'$(NetCoreAppImplicitPackageVersion)' == ''">1.1.0</NetCoreAppImplicitPackageVersion>
+      </PropertyGroup>
+    </When>
+    
+    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(RuntimeFrameworkVersion)' != ''">
+      <PropertyGroup>
+        <NetCoreAppImplicitPackageVersion Condition="'$(NetCoreAppImplicitPackageVersion)' == ''">$(RuntimeFrameworkVersion)</NetCoreAppImplicitPackageVersion>
+      </PropertyGroup>
+    </When>
+  </Choose>
+  
+  
+  <!-- Remove the implicit PackageReference items if there are non-implicit PackageReference items with the same identity.
+       Do this during evaluation instead of adding a target because Visual Studio only uses evaluation to get the
+       PackageReferenceitems to restore. -->
+  <PropertyGroup Condition="'$(DisableImplicitFrameworkReferences)' != 'true'">
+    <_ImplicitPackageName Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard'">NETStandard.Library</_ImplicitPackageName>
+    <_ImplicitPackageName Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">Microsoft.NETCore.App</_ImplicitPackageName>
+  </PropertyGroup>
+  
+  <ItemGroup Condition="'$(_ImplicitPackageName)' != ''">
+    <!-- Filter PackageReference to items where the ItemSpec matches the implicit package name, and add IsImplicitlyDefined metadata
+         for items that don't have it-->
+    <_ImplicitPackageReferenceCheck
+        Include="@(PackageReference->WithMetadataValue('Identity', '$(_ImplicitPackageName)'))">
+      <IsImplicitlyDefined Condition="'%(IsImplicitlyDefined)' != 'true' ">false</IsImplicitlyDefined>
+    </_ImplicitPackageReferenceCheck>
+    
+    <!-- Now filter down to an item with just the implicit reference and another one with just the overriding reference -->
+    <_ImplicitPackageReference Include="@(_ImplicitPackageReferenceCheck->WithMetadataValue('IsImplicitlyDefined', 'true'))"/>
+    <_OverridingPackageReference Include="@(_ImplicitPackageReferenceCheck->WithMetadataValue('IsImplicitlyDefined', 'false'))"/>
+
+    <!-- Remove both the implicit and the override item, if there was both an implicit and an override item -->
+    <PackageReference Remove="@(PackageReference->WithMetadataValue('Identity', '$(_ImplicitPackageName)'))"
+                      Condition="'@(_ImplicitPackageReference)' != '' And '@(_OverridingPackageReference)' != ''"
+                      />
+
+    <!-- Add the override item back -->
+    <PackageReference Include="@(_OverridingPackageReference)"
+                      Condition="'@(_ImplicitPackageReference)' != '' And '@(_OverridingPackageReference)' != ''" />
```

Comment:
I can't either. In a target, you'd batch to evaluate the condition multiple times but that wouldn't work here. We'll have to see if this comes up often enough to worry about providing convenience for.